src/LinearSolver.jl

```diff
@@ -83,23 +83,32 @@ PardisoSolver() = PardisoSolver(MTYPE_REAL_NON_SYMMETRIC,
 PardisoSolver constructor overloading with default values.
 Returns a PardisoSolver given its matrix type.
 """
-PardisoSolver(mtype) = 
+PardisoSolver(mtype::Int) = 
     PardisoSolver(mtype, new_iparm(), MSGLVL_QUIET, new_pardiso_handle())
 
+PardisoSolver(::Type{<:AbstractSparseMatrix{Tv,Ti}}) where {Tv,Ti}= 
+    PardisoSolver(GridapPardiso.MTYPE_REAL_NON_SYMMETRIC)
+
+PardisoSolver(::Type{SymSparseMatrixCSR{Bi,Tv,Ti}}) where {Bi,Tv,Ti}= 
+    PardisoSolver(GridapPardiso.MTYPE_REAL_SYMMETRIC_INDEFINITE)
+
+PardisoSolver(op::AffineFEOperator) = 
```

Comment:
Instead of implementing it only for `AffineFEOperator`, I would do it for any `FEOperator` like this:

```julia
PardisoSolver(op::FEOperator) = PardisoSolver(get_matrix_type(op))
```

And then, implement 

```julia
function get_matrix_type(op::FEOperator)
  @abstractmethod
end

function get_matrix_type(op::AffineFEOperator)
  typeof(get_matrix(op))
end
```

These two last functions will be eventually moved to the Gridap repo.

Reply:
Commit d88a2a0 solves this comment